pkg/github/pagination.go

```diff
@@ -0,0 +1,44 @@
+package github
+
+import (
+	"cmp"
+	"fmt"
+	"sort"
+
+	"github.com/google/go-github/v61/github"
+)
+
+func Paginate[T cmp.Ordered](f func(x func(T), opts *github.ListOptions) (*github.Response, error)) ([]T, error) {
```

Comment:
Adding `accum` seems a bit verbose to the func signature and harder to understand. Would this work?

```go
func Paginate[T cmp.Ordered](f func(opts *github.ListOptions) ([]T, *github.Response, error)) ([]T, error) {
  allItems := make([]T, 32)
  for {
    // get the current page
    items, resp, err := f(opts)
    // handle error and stuff
    allItems = append(allItems, items)
    // move to next page if any
  }
  // sort
  return allItems, nil
}
```

Reply:
I had that originally, and I agree that `accum` is weird. The problem with the non-accum approach is that we end up allocating a lot of slices and maps (since we need the map to de-dupe since pagination can result in duplicates when new entries are added during pagination).

```
allItems = append(allItems, items)
```

needs to be:

```
for _, v := range items {
  if _, ok := allItems[v]; ok {
    continue
  }
  allItems[v] = struct{}{}
}
```

Reply:
I dropped accum entirely and allow it to be local now.